microsoft-edge/webview2/gettingstarted/wpf.md

```diff
@@ -0,0 +1,291 @@
+
+```csharp
+public MainWindow()
+{
+    InitializeComponent();
+    webView.NavigationStarting += EnsureHTTPS;
+    InitializeAsync();
+}
+
+async void InitializeAsync()
+{
+    await webView.EnsureCoreWebView2Async(null);
+}
+```
+
+2. After **CoreWebView2** is initialized, register an event handler to respond to **Web Message Received**. In this example, the host does three things:
+    1. Receives a web message (the URI) from the web content using [TryGetWebMessageAsString](../reference/dotnet/0-9-494/microsoft-web-webview2-core-corewebview2webmessagereceivedeventargs#trygetwebmessageasstring).
+    2. Updates the text in the search bar to the URI
+    3. Returns the message (the URI) back to the web content using [PostWebMessageAsString](../reference/dotnet/0-9-494/microsoft-web-webview2-core-corewebview2#postwebmessageasstring)
+
+In **MainWindow.xaml.cs** update `InitializeAsync` and define `UpdateSearchBar` as follows:
+
+```csharp
+async void InitializeAsync()
+{
+    await webView.EnsureCoreWebView2Async(null);
+    webView.CoreWebView2.WebMessageReceived += UpdateSearchBar;
+}
+
+void UpdateSearchBar(object sender, CoreWebView2WebMessageReceivedEventArgs args)
+{
+    String Uri = args.TryGetWebMessageAsString();
+    searchBar.Text = Uri;
+    webView.CoreWebView2.PostWebMessageAsString(Uri);
+}
+
+```
+
+3. In order for the WebView to send and respond to the web message, after **CoreWebView2** is initialized, the host:
+
+    1. Injects a script to the web content that registers a handler to print message from the host
+    2. Injects a script to the web content that posts the URL to the host
+
+In **MainWindow.xaml.cs** update `InitializeAsync` as follows:
+
+```csharp
+async void InitializeAsync()
+{
+    await webView.EnsureCoreWebView2Async(null);
+    webView.CoreWebView2.WebMessageReceived += UpdateSearchBar;
+
+    await webView.CoreWebView2.AddScriptToExecuteOnDocumentCreatedAsync("window.chrome.webview.postMessage(window.document.URL);");
+    await webView.CoreWebView2.AddScriptToExecuteOnDocumentCreatedAsync("window.chrome.webview.addEventListener(\'message\', event => alert(event.data));");
+}
+```
+
+Press F5 to build and run the app. Now the search bar displays the URI in the WebView and when you successfully navigate to a new URI, the WebView alerts the user of the URI displayed in the WebView.
+
+![searchbar](media/wpf-gettingstarted-searchbar.png)
+
+Congratulations, you've just built your first WebView2 app!
+
+## Next Steps
+
+There are plenty of WebView2 functionalities that are not covered in this walkthrough.
+
+To learn more:
+
+* Please explore [API reference](../reference/dotnet/0-9-494-reference-webview2.md) for detailed information about each API.  
+
+## Feedback
+
+Help us build a richer WebView2 experience by sharing your feedback! Visit our [feedback repo](https://aka.ms/webviewfeedback) to submit feature requests or bug reports or search for known issues.
```

Comment:
Can you add a See Also section with 3-5 links to related content please?

Reply:
We have a Next Steps section. We will link to the sample apps and reference when they are published should be a simple change. They will be published tomorrow morning